extensions/elasticsearch-rest-client/runtime/src/main/java/io/quarkus/elasticsearch/restclient/lowlevel/runtime/RestClientBuilderHelper.java

```diff
@@ -77,7 +81,15 @@ public HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder httpCli
                     httpClientBuilder.setSSLStrategy(NoopIOSessionStrategy.INSTANCE);
                 }
 
-                return httpClientBuilder;
+                // Apply configuration from RestClientBuilder.HttpClientConfigCallback implementations annotated with ElasticsearchClientConfig
+                HttpAsyncClientBuilder result = httpClientBuilder;
```

Comment:
This line didn't improve readibility, can you get rid of it ?

Reply:
Do you mean the comment line?

Reply:
No, this one:

```
HttpAsyncClientBuilder result = httpClientBuilder;
```

I don't think copying the `httpClientBuilder` inside a `result` variable didn't adds anything here, but it's nitpicking ;)

Reply:
Do you suggest I update the original "httpClientBuilder" variable with the value returned from the HttpClientConfigCallback implementations within the for loop or, I should discard the return values of user supplied HttpClientConfigCallback implementations and just return the httpClientBuilder?

Reply:
Sorry, I read to quickly the code and didn't notice that you need to re-assign the httpClientBuilder to itself then return it.
OK for me then